framework/src/main/java/org/checkerframework/common/basetype/BaseTypeValidator.java

```diff
@@ -701,4 +703,51 @@ public boolean areBoundsValid(
 
         return true;
     }
+    /**
+     * Validate if the annotations on wildcard are permitted by @TargetLocations. Report an error if
+     * the actual use of this annotation is not listed in the declared TypeUseLocations
+     * in @TargetLocation.
+     *
+     * @param type the type to check
+     * @param tree the tree of this type
+     */
+    protected void validateWildCardTargetLocation(AnnotatedWildcardType type, Tree tree) {
+        if (tree.getKind() == Tree.Kind.SUPER_WILDCARD) {
+            for (AnnotationMirror am : type.getSuperBound().getAnnotations()) {
+                List<TypeUseLocation> locations =
+                        visitor.AnnoToTargetLocations.get(AnnotationUtils.annotationName(am));
+                // @Target({ElementType.TYPE_USE})} together with no @TargetLocations(...) means
+                // that the qualifier can be written on any type use
+                if (locations == null
+                        || locations.contains(TypeUseLocation.EXPLICIT_LOWER_BOUND)
+                        || locations.contains(TypeUseLocation.LOWER_BOUND)
+                        || locations.contains(TypeUseLocation.ALL)) {
+                    continue;
+                }
+
```

Comment:
`@A ? super @ B Bottom` // @ A is an explicit upper bound?
`? super @ B Bottom` // implicit upper bound
`type.getExtendsBound()`

just the wildcard case `@A ?`